### ISSUE TITLE
Inherit SideTitles text color from app theme

### ISSUE THREAD

**Issue by enricobenedos** [imaNNeoFighT/fl_chart#269]
**Is your feature request relasted to a problem? Please describe.**
Good Morning,
I'm implementing the darkTheme on my app. Differently from a lot of other widgets, chart SideTitles does not inherit the default `primaryTextTheme` color. 

Actually I need to invoke `Theme.of(context)` and get the desired app wide text color
```dart                      
                     SideTitles(
                        textStyle: TextStyle(
                            color:
                                Theme.of(context).primaryTextTheme.title.color,
                            fontSize: 11.0),
                        reservedSize: 35.0,
                        showTitles: true,
                        interval: yAxisInterval > 1 ? yAxisInterval : 15,
                        getTitles: (value) {
                          return '${value.toStringAsFixed(0)} m';
                        },
                      ),
```
This workaround need to redefine also the `fontSize` property because it overrides the standard `SideTitles` one
**Describe the solution you'd like**
Can it be an idea to automatically inherit the text color from the application theme?

By me the change can be to simply remove the forced black color, this permit to automatically inherit from the app theme
```dart
  const SideTitles({
    this.showTitles = false,
    this.getTitles = defaultGetTitle,
    this.reservedSize = 22,
    this.textStyle = const TextStyle(
      //color: Colors.black,
      fontSize: 11,
    ),
    this.margin = 6,
    this.interval = 1.0,
    this.rotateAngle = 0.0,
  });
```

Thank you

**Comment by imaNNeoFighT** [comment]
Hi,
Thanks for letting us know about this issue,
we will consider it.
Thanks for reporting!

**Comment by imaNNeoFighT** [comment]
Hi.
We have just released [0.36.4](https://github.com/imaNNeoFighT/fl_chart/releases/tag/0.36.4) version.
Now we handle `textStyle` exactly like Flutter's [Text](https://api.flutter.dev/flutter/widgets/Text-class.html) widget. All [TextStyles](https://api.flutter.dev/flutter/painting/TextStyle-class.html) are theme-aware and nullable.
It means if you don't provide anything we use Theme directly.
Or if you provide a customized [TextStyles](https://api.flutter.dev/flutter/painting/TextStyle-class.html) we try to merge it with the theme's textStyle, instead of replacing it.

Thanks for your suggestion and contribution.
Cheers!

**Comment by 2shrestha22** [comment]
But can cause unexpected behavior since `fl_chart: ^0.36.3` automatically satisfies to 0.36.4 and hence in result save piece of code can work in on system and won't in another.
